src/content/docs/cheerpj3/11-guides/CheerpJ-explanation.mdx

```diff
@@ -0,0 +1,81 @@
+
+## CheerpJ components
+
+CheerpJ is made of the following building blocks:
+
+- A full Java runtime environment with its implementation of the JVM and a Java-to-JavaScript JIT compiler.
+- A virtualized window manager
+- A virtualized file system
+- A module for networking support
+
+![CheerpJ 3.0 architecture diagram](/cheerpj3/assets/cheerpj-3-arch.png)
+
+### The CheerpJ Java Runtime Environment
+
+The magic behind CheerpJ is [Cheerp], which was used for compiling a full Java SE 8 runtime based on OpenJDK. This runtime was originally written in C++ and compiled to WebAssembly and JavaScript, making it 100% browser compatible. The architecture is designed to support multiple versions of Java, as well as custom runtimes. Future versions of CheerpJ will support Java 11, and newer LTS Java versions.
+
+The most important component of the CheerpJ Java runtime environment is its JVM implementation. CheerpJ's implementation of the JVM is made of an _interpreter_ and a _Just In Time compiler_ (JIT) that work in conjunction in a 2-tier manner:
+
+1. The Java byte code runs within an interpreter.
+2. Then this byte code is Just-In-Time compiled to optimized JavaScript.
+
+The interpreter does not only deal with initialization and rarely-used code, but also gathers necessary information for JIT-ting. The generated code is very efficient, and the internal optimizer can, among other things, inline and devirtualize calls, which is extremely important for a language such as Java.
+
+![CheerpJ 3.0 JRE diagram](/cheerpj3/assets/JRE.png)
+
+Alongside being able to run Java applications in the browser, CheerpJ runtime also provides advanced, bidirectional Java-JavaScript interoperability. Meaning you can access DOM from Java by [implementing `native` methods directly in JavaScript](/cheerpj3/guides/Implementing-Java-native-methods-in-JavaScript). You can also interact with Java methods, objects and arrays directly from JavaScript by using the new [`cheerpjRunLibrary` API](/cheerpj3/reference/cheerpjRunLibrary).
+
+### The virtualized window manager
+
+What would be of an application without its graphical UI? CheerpJ's window manager supports Java AWT/Swing by converting windows to a hierarchy of HTML elements and HTML5 canvases.
+
+Swing applications, will render exactly as they do in native. Swing Look&Feel is also supported, including 3rd party ones. Multi-window applications are supported, with keyboard focus being managed as expected. Integration with the system clipboard can be enabled via an initialization option.
+
+### The virtualized file system
+
+CheepJ provides [multiple filesystem backends](cheerpj3/guides/File-System-support) to accommodate different application needs, including access to server-hosted files and persistent local storage.
+
+![](/cheerpj2/assets/filesystem.png)
+
+You can learn more about CheerpJ's filesystem at our [files and filesystems guide].
```

Comment:
s/at/in

I think this link, and the link in the networking section, would be better off as LinkButtons with a right arrow icon